docs/source/common/optimizers.rst

```diff
@@ -592,3 +593,21 @@ Here we configure gradient clipping differently for optimizer B.
             self.clip_gradients(
                 optimizer, gradient_clip_val=gradient_clip_val * 2, gradient_clip_algorithm=gradient_clip_algorithm
             )
+
+
+Total Stepping Batches
+======================
+
+You can use built-in trainer property :paramref:`~pytorch_lightning.trainer.trainer.Trainer.estimated_stepping_batches` to compute
+total number of stepping batches for the complete training. The property is computed considering gradient accumulation factor and
+distributed setting into consideration so you don't have to derive it manually. One good example where this can be helpful is while using
+:class:`~torch.optim.lr_scheduler.OneCycleLR` scheduler, which requires pre-computed ``total_steps`` during initialization.
+
+.. code-block:: python
+
+    def configure_optimizers(self):
+        optimizer = ...
+        scheduler = torch.optim.lr_scheduler.OneCycleLR(
+            optimizer, max_lr=1e-3, total_steps=self.trainer.estimated_stepping_batches
+        )
+        return [optimizer], [scheduler]
```

Comment:
Hi!

I think this code example may be misleading, bacause (if I get it right) the default scheduler interval is 'epoch', so here the scheduler would only update every epoch (and wouldn't complete the cycle, because the total_steps is set to estimated_stepping_batches, not number of epochs). I tested it in [Colab](https://colab.research.google.com/drive/1fDL6d1whBex9qf4tfELBCdLgkmPf6IXZ?usp=sharing)

Should it be like this?
```
    return [optimizer], [{"scheduler": scheduler, "interval": "step"}]
```

Reply:
Sounds good to me. Would you like to send a PR updating this example?